### DISCUSSION TITLE
Um bom front-end precisa ser expert na criação de layouts?

### DISCUSSION THREAD

**Discussion by Marcelosilva10** [frontendbr/forum#176] [Discussão]
Essa e uma dúvida que sempre bate em minha cabeça sabe, vi aqui a carreira de front end excelente conteúdo e muito rico, até agradeço a todos que sempre me ajudam aqui vejo os que manjam aqui se encontram...

Tenho um seguinte duvida como falei para ser um bom front end precisa manja de criação de layout desde do passo iniciais ou não precisa saber html, css e javascript, pode ser até uma pergunta infantil da minha parte acho que aqui posso buscar ajuda.

Queria saber a opinião de vocês?
Ótimo final de semana a todos e bom divertimento!

**Comment by felipefialho** [comment]
Sua dúvida é com relação a criação do design do layout ou sobre o desenvolvimento da interface?

**Comment by Marcelosilva10** [comment]
Design do layout, isso fica ao papel do Design?

Desenvolvimento de Interface fica no papel do Front end?

Qual seria a diferença?

**Comment by felipefialho** [comment]
Em tese é isso @Marcelosilva10. O desenvolvimento do layout (a imagem) fica a cargo do designer. E o desenvolvimento da interface (o código) a cargo do front-ender.

**Comment by felipecodes** [comment]
@Marcelosilva10 recomendo a leitura desse artigo => 
https://diessi.ca/blog/a-fantastica-diversidade-do-front-end/

**Comment by kandebonfim** [comment]
YEP!

**Comment by Marcelosilva10** [comment]
@felipoliveira obrigado Felipe!

**Comment by felipeuntill** [comment]
Esta questão é fortemente baseada em opiniões uma vez que a atuação de um profissional front-end é relativamente abrangente no "Brasil", mas considerando as minhas experiências posso afirmar que ter noções de design é realmente um diferencial. 

Rotineiramente sempre existem situações onde o layout não atende os requisitos e ter um recurso capaz de intervir sem a ajuda de um designer, permite que o time seja ágil. 

As habilidades de criação não precisam ser extraordinárias mas um recurso com bom gosto e noções de tipografia, seguidas de experiência no uso dos principais editores de imagens do mercado são sim um enorme **Plus**.

Generalizar é sempre um erro mas nunca vi um frontender sem noções básicas de design que fosse detalhista e minucioso.

**Comment by kvnol** [comment]
Reforçando o que o @felipeuntill disse, você como front-end não precisa ser o mestre do gestalt ou aquele _cara que chora quando vê um quadro_ (peguei essa referência no Dev Na Estrada), mas é importante ter um olho crítico e sinceridade para dizer "isso está bem ruim".

Eu comecei a desenvolver dos dois lados, criando layouts e desenvolvendo os mesmos no front, hoje meu foco é mais front, mas se tu trampa em pequenas agências/empresas, é quase certeza que você terá que fazer os dois (e às vezes até o back-end), por isso não é necessário ser especialista, mas conhecer é importantíssimo.

> Meu sonho é um dia trabalhar com um designer na equipe. — LISPECTOR, Clarice

**Comment by willycamargo** [comment]
Acho que a issue #32 pode te ajudar a clarear um pouco também.

**Comment by rubenmarcus** [comment]
É bom sim, eu como vim e gosto mais de Design acho que é essencial. Teve uma discussão recente no grupo do facebook de um cara falando que não fazem mais sites como o EYE4U, e um estudio famoso lá que tem o site em Flash.
Acho que não precisa ser assim, mas não precisa ser igual 90% dos portfolios que são onepage.

Saber o básico sobre design para ter autonomia é essencial. Ja vi muito designer que sabe pouco sobre usabilidade e faz layouts lindos, mas horríveis para codificar. Acho que essa necessidade é a mais útil ter noções de design ( Tipografia,Cores,AI,Estilos,tendencias etc) e UI/UX é essencial.

Mas tem o foco de cada um... como reportaram na issue acima.
